components/service/nimbus/src/main/java/mozilla/components/service/nimbus/Nimbus.kt

```diff
@@ -1,554 +1,85 @@
-        nimbus.getAvailableExperiments()
-
-    override fun getExperimentBranch(experimentId: String): String? {
-        recordExposure(experimentId)
-        return nimbus.getExperimentBranch(experimentId)
-    }
-
-    @WorkerThread
-    override fun getExperimentBranches(experimentId: String): List<Branch>? = withCatchAll {
-        nimbus.getExperimentBranches(experimentId)
-    }
-
-    override fun updateExperiments() {
-        fetchScope.launch {
-            fetchExperimentsOnThisThread()
-            applyPendingExperimentsOnThisThread()
-        }
-    }
-
-    // Method and apparatus to catch any uncaught exceptions
-    @SuppressWarnings("TooGenericExceptionCaught")
-    private fun <R> withCatchAll(thunk: () -> R) =
-        try {
-            thunk()
-        } catch (e: Throwable) {
-            try {
-                errorReporter(e)
-            } catch (e1: Throwable) {
-                logger.error("Exception calling rust", e)
-                logger.error("Exception reporting the exception", e1)
-            }
-            null
-        }
-
-    override fun initialize() {
-        dbScope.launch {
-            initializeOnThisThread()
-        }
-    }
-
-    @WorkerThread
-    @VisibleForTesting(otherwise = VisibleForTesting.PRIVATE)
-    internal fun initializeOnThisThread() = withCatchAll {
-        nimbus.initialize()
-    }
-
-    override fun fetchExperiments() {
-        fetchScope.launch {
-            fetchExperimentsOnThisThread()
-        }
-    }
-
-    @WorkerThread
-    @VisibleForTesting(otherwise = VisibleForTesting.PRIVATE)
-    internal fun fetchExperimentsOnThisThread() = withCatchAll {
-        try {
-            nimbus.fetchExperiments()
-            notifyObservers { onExperimentsFetched() }
-        } catch (e: NimbusErrorException.RequestError) {
-            logger.info("Error fetching experiments from endpoint: $e")
-        } catch (e: NimbusErrorException.ResponseError) {
-            logger.info("Error fetching experiments from endpoint: $e")
-        }
-    }
-
-    override fun applyPendingExperiments() {
-        dbScope.launch {
-            applyPendingExperimentsOnThisThread()
+    errorReporter: ErrorReporter = loggingErrorReporter,
+    private val observable: Observable<NimbusInterface.Observer> = ObserverRegistry()
+) : ApplicationServicesNimbus(
+    context = context,
+    appInfo = appInfo,
+    server = server,
+    deviceInfo = NimbusDeviceInfo(
+        localeTag = Locale.getDefault().getLocaleTag()
+    ),
+    observer = Observer(observable),
+    delegate = NimbusDelegate(
+        dbScope = CoroutineScope(Executors.newSingleThreadExecutor(
+            NamedThreadFactory("NimbusDbScope")
+        ).asCoroutineDispatcher()),
+        fetchScope = CoroutineScope(Executors.newSingleThreadExecutor(
+            NamedThreadFactory("NimbusFetchScope")
+        ).asCoroutineDispatcher()),
+        errorReporter = errorReporter,
+        logger = { logger.info(it) }
```

Comment:
nit: You used to log errors here in various functions, now it's all infos. Intentional?

Reply:
Everything that used to report an error via the `logger.error()` is now doing it via an `errorReporter`. 

The only exception to this is when an exception is caught coming from `errorReporter`. If `errorReporter` is itself erroring, I'm guessing nimbus won't be the only one finding out about it.

If/when `Nimbus` does log non-error messages I'd rather do it by `info` than by `error`.